### PR TITLE
ENH: PointSet metric and registration method supports perspective proj.

### PR DESCRIPTION
**Pull request by nhjohnston**
Extended wrapping configuration for PointSetToPointSetMetric and
for PointSetToPointSetRegistrationMethod to support perspective
transform for 2D to 3D registration.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Modules/Registration/Common/wrapping/itkPointSetToPointSetMetric.wrap [hunk 1]
```diff
@@ -7,4 +7,9 @@ itk_wrap_class("itk::PointSetToPointSetMetric" POINTER_WITH_2_SUPERCLASSES)
       itk_wrap_template("PS${ITKM_${t}}${d}" "itk::PointSet< ${ITKT_${t}},${d} >, itk::PointSet< ${ITKT_${t}},${d} >")
     endforeach()
   endforeach()
-itk_end_wrap_class()
+  if(d3 AND d2)
+    foreach(t ${types})
+      itk_wrap_template("PS${ITKM_${t}}23" "itk::PointSet< ${ITKT_${t}},2 >, itk::PointSet< ${ITKT_${t}},3 >")
```
Comment (dzenanz) [inline Modules/Registration/Common/wrapping/itkPointSetToPointSetMetric.wrap:12]: Should 3D->2D case be wrapped too? This does it for 2D->3D.

File: Modules/Registration/Common/wrapping/itkPointSetToPointSetRegistrationMethod.wrap [hunk 2]
```diff
@@ -6,9 +6,15 @@ UNIQUE(types "${WRAP_ITK_REAL};D")
...
   foreach(d ${ITK_WRAP_DIMS})
-    foreach(t ${WRAP_ITK_REAL})
+    foreach(t ${types})
       itk_wrap_template("REG${ITKM_${t}}${d}${ITKM_${t}}${d}"
                         "itk::PointSet< ${ITKT_${t}}, ${d} >, itk::PointSet< ${ITKT_${t}}, ${d} >")
     endforeach()
   endforeach()
+  if(d3 AND d2)
+    foreach(t ${types})
+      itk_wrap_template("REG${ITKM_${t}}2${ITKM_${t}}3"
```
Comment (dzenanz) [inline Modules/Registration/Common/wrapping/itkPointSetToPointSetRegistrationMethod.wrap:16]: Also 3->2?

File: Modules/Registration/Common/wrapping/itkPointSetToPointSetMetric.wrap [hunk 3]
```diff
@@ -12,4 +12,4 @@ itk_wrap_class("itk::PointSetToPointSetMetric" POINTER_WITH_2_SUPERCLASSES)
       itk_wrap_template("PS${ITKM_${t}}23" "itk::PointSet< ${ITKT_${t}},2 >, itk::PointSet< ${ITKT_${t}},3 >")
     endforeach()
   endif()
-itk_end_wrap_class()
\ No newline at end of file
+itk_end_wrap_class()
```
Comment (dzenanz) [inline Modules/Registration/Common/wrapping/itkPointSetToPointSetMetric.wrap:15]: These commits should be squashed, upon merge if not before.

File: Modules/Registration/Common/wrapping/itkPointSetToPointSetMetric.wrap (hunk 1 above)
Comment (aylward) [reply Modules/Registration/Common/wrapping/itkPointSetToPointSetMetric.wrap:12]: I believe only 2D->3D can be solved with a pair of images.  There is no way to solve 3D->2D.    That is, the fixed point set must be 2D, and the perspective transform moves the 3D points into 2D space and the metric is a 2D to 2D metric.   Determining how to project 2D points into 3D space requires multiple projection images.

File: Modules/Registration/Common/wrapping/itkPointSetToPointSetRegistrationMethod.wrap (hunk 2 above)
Comment (aylward) [reply Modules/Registration/Common/wrapping/itkPointSetToPointSetRegistrationMethod.wrap:16]: I think only fixed 2D and moving 3D is possible using the perspective transform.

File: Modules/Registration/Common/wrapping/itkPointSetToPointSetMetric.wrap (hunk 3 above)
Comment (aylward) [reply Modules/Registration/Common/wrapping/itkPointSetToPointSetMetric.wrap:15]: Agree!

**Comment by PranjalSahu** [comment]
LGTM

**Review by dzenanz — APPROVED** [review]

**Review by PranjalSahu — APPROVED** [review]

**Pull request merged**